src/pages/_app.tsx

```diff
@@ -1,23 +1,57 @@
 import type { AppProps } from 'next/app'
 import NextHead from 'next/head'
 import { FixedGlobalStyle, ThemedGlobalStyle } from '../theme'
-import { StarknetProvider } from '@starknet-react/core'
+import { StarknetProvider, useStarknet, useStarknetCall } from '@starknet-react/core'
 import { BigNumber } from 'bignumber.js'
 import React from 'react'
-import {S2MTransactionManagerProvider} from "../providers/transaction";
+import { S2MTransactionManagerProvider } from '~/providers/transaction'
+import { useGameContract } from '~/hooks/game'
+import Popups from '~/components/Popups'
+import { AppWrapper } from '~/components/Core/AppWrapper'
+import ConnectArgentx from '~/pages/loginOrGenerate'
+import useGeneratePlanet from '~/hooks/calls/useGeneratePlanet'
 
-function MyApp({ Component, pageProps }: AppProps) {
+const AuthController = ({ Component, pageProps }: AppProps) => {
+  const { account } = useStarknet()
+  const { contract: gameContract } = useGameContract()
+  // Not sure about the returned result of this call, if 0 then no planet?
+  const { data, error } = useStarknetCall({
+    contract: gameContract,
+    method: 'owner_of',
+    args: [account],
+  })
+  const generatePlanet = useGeneratePlanet()
+
+  // useMemo(() => {
```

Comment:
not sure about result here, so for now I let just the call without check if you have the planet

Reply:
Yes, if planet_id === 0, then planet is not generated yet. Otherwise, planet is generated